mantidproject/mantid
# Indirect Use new plotting widget

~~Merge after #26377~~
--------------------------

**Description of work.**
This PR replaces the old plotting widgets seen on the Indirect interfaces. Instead, a new widget is used which is tested and will improve interface consistency.

**To test:**
MantidPlot and Workbench
1. `Interfaces`->`Indirect`->`Data Corrections`-> All Tabs
2. Load and enter the data below for the respective tabs.
3. You should try observe several things:
- On ContainerSubtraction and ApplyAbsorptionCorrections: do `PlotCurrentPreview` and make sure it works with errorBars being turned off and on in the Settings GUI.
- On CalcPaalmanPings and MonteCarlo: Run everything and then try `Plot Wavelength` and `Plot Angle`. These plots should also work with the error bars option in the Settings GUI.
                     
This test covers any new functionality added across the Indirect interfaces.

**Container Subtraction:**
Sample: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460924/irs26176_graphite002_red.zip)
Container: [irs26174_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460923/irs26174_graphite002_red.zip)

**Calculate Paalman Pings:**
Sample: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460924/irs26176_graphite002_red.zip)
Sample Thickness: `0.1cm`
Chemical Formula: `H2-O`

**Calculate Monte Carlo Absorption:**
Sample: `26176` (on the system)
Container: `26174` (on the system)
Shape: `FlatPlate`
Sample and Container Height & width: `1.0`
Sample and Container Thickness: `0.1`
Sample mass density: `1`
Sample Formula: `H2-O`
Container mass density: `6`
Container Formula: `V`
It may be a long wait when running! The output workspace named 

**Apply Absorption Corrections:**
Sample: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460924/irs26176_graphite002_red.zip)
Corrections: [irs26176_graphite002_FlatPlate_MC_Corrections.zip](https://github.com/mantidproject/mantid/files/2464168/irs26176_graphite002_FlatPlate_MC_Corrections.zip)
Tick `Use Container`
Container: [irs26174_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460923/irs26174_graphite002_red.zip)

Fixes #26511

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
